### PR TITLE
Run2-alca115 Update the code of AlCaIsoTracksFilter to check source of losing events

### PR DESCRIPTION
**Pull request by bsunanda**
Selection of the filter and downstream tree maker is made as similar as possible with the filter having somewhat looser selection criteria

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Calibration/HcalAlCaRecoProducers/plugins/AlCaIsoTracksFilter.cc [hunk 1]
```diff
@@ -84,19 +84,21 @@ class AlCaIsoTracksFilter : public edm::stream::EDFilter<edm::GlobalCache<AlCaIs
...
-  double                        a_mipR_, a_coneR_, a_charIsoR_;
-  double                        pTrackMin_, eEcalMax_, eIsolate_;
-  double                        pTrackLow_, pTrackHigh_;
-  int                           preScale_;
-  unsigned int                  nRun_, nAll_, nGood_, nRange_;
-  edm::InputTag                 triggerEvent_, theTriggerResultsLabel;
-  edm::InputTag                 labelGenTrack_, labelRecVtx_;
-  edm::InputTag                 labelEB_, labelEE_, labelHBHE_;
+  HLTConfigProvider              hltConfig_;
+  std::vector<std::string>       HLTNames_;
```
Comment (ghellwig) [inline Calibration/HcalAlCaRecoProducers/plugins/AlCaIsoTracksFilter.cc]: I see that you do not make this one const, while you changed `trigNames_` to a const member.
But now I am failing to see where `HLTNames_` is used at all. Could this maybe just go?

File: Calibration/HcalAlCaRecoProducers/plugins/AlCaIsoTracksFilter.cc [hunk 2]
```diff
@@ -407,11 +411,44 @@ void AlCaIsoTracksFilter::endRun(edm::Run const& iRun, edm::EventSetup const&) {
...
+  desc.add<double>("maximumEcalEnergy",2.0);
+  // following 3 parameters are for isolation cuts and described in the code
+  desc.add<double>("maxTrackP",8.0);
+  desc.add<double>("slopeTrackP",0.05090504066);
+  desc.add<double>("isolationEnergy",10.0);
+  // Prescale events only containing isolated tracks in the range
+  desc.add<double>("momentumRangeLow",20.0);
+  desc.add<double>("momentumRangeHigh",40.0);
+  desc.add<int>("preScaleFactor",1);
+  descriptions.add("AlcaIsoTracksFilter",desc);
```
Comment (ghellwig) [inline Calibration/HcalAlCaRecoProducers/plugins/AlCaIsoTracksFilter.cc]: please change that to "alcaIsoTracksFilter" and accordingly below

**Review by ghellwig — COMMENTED** [review]
looks good, i.e. only minor comments

**Comment by bsunanda** [comment]
@ghellwig I could not make theTrackQuality_ as a const because it is used as an argument in a call to a method of Calibration/IsolataedParticles where it is a std::string - I shall take a look at that part of the code later on.

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by ghellwig** [comment]
+1
thanks @bsunanda

**Comment by fabiocos** [comment]
+1

**Pull request merged**